model/src/application/interaction/application_command/option.rs

```diff
@@ -181,129 +182,157 @@ impl<'de> Deserialize<'de> for CommandDataOption {
+                                CommandOptionValue::Role(id.cast())
+                            } else {
                                 return Err(DeError::invalid_type(
-                                    make_unexpected(&other),
-                                    &"string",
+                                    make_unexpected(&val),
+                                    &"role id",
                                 ));
                             }
                         }
-                    }
-                    CommandOptionType::SubCommand => CommandOptionValue::SubCommand(options),
-                    CommandOptionType::SubCommandGroup => {
-                        CommandOptionValue::SubCommandGroup(options)
-                    }
-                    CommandOptionType::User => {
-                        let val = value_opt.ok_or_else(|| DeError::missing_field("value"))?;
 
-                        if let ValueEnvelope::Id(id) = val {
-                            CommandOptionValue::User(id.cast())
-                        } else {
-                            return Err(DeError::invalid_type(make_unexpected(&val), &"user id"));
+                        CommandOptionType::String => {
+                            let val = value_opt.ok_or_else(|| DeError::missing_field("value"))?;
+
+                            match val {
+                                ValueEnvelope::String(s) => CommandOptionValue::String(s),
+                                ValueEnvelope::Id(id) => {
+                                    CommandOptionValue::String(id.get().to_string())
+                                }
+                                other => {
+                                    return Err(DeError::invalid_type(
+                                        make_unexpected(&other),
+                                        &"string",
+                                    ));
+                                }
+                            }
+                        }
+                        CommandOptionType::SubCommand => CommandOptionValue::SubCommand(options),
+                        CommandOptionType::SubCommandGroup => {
+                            CommandOptionValue::SubCommandGroup(options)
+                        }
+                        CommandOptionType::User => {
+                            let val = value_opt.ok_or_else(|| DeError::missing_field("value"))?;
+
+                            if let ValueEnvelope::Id(id) = val {
+                                CommandOptionValue::User(id.cast())
+                            } else {
+                                return Err(DeError::invalid_type(
+                                    make_unexpected(&val),
+                                    &"user id",
+                                ));
+                            }
                         }
                     }
                 };
 
                 Ok(CommandDataOption {
+                    focused,
```

Comment:
This information is not useful, remove it from the API (encoded in `CommandOptionValue`)